util/dvsim/dvsim.py

```diff
@@ -158,6 +161,45 @@ def sigint_handler(signal_received, frame):
     exit(1)
 
 
+def copy_repo_to_scratch_area(src, dest, dry_run):
+    '''Copy over the repo to the scratch area.
+
+    This is done if the env variable DVSIM_REMOTE_DISPATCH is set and the
+    scratch_root is not set to default. In corp environments, the jobs may be
+    dispatched to the remote compute machines which have access to dest
+    but not the src area. The rsync utility provides the ability to
+    specify a file containing exclude patterns, which can be used to skip
+    copying over the build directories and other temporary files.
+    '''
+
+    rsync_cmd = "rsync --archive --update --executability --verbose "
+    # Supply `.gitignore` from the src area to skip temp files.
+    ignore_patterns_file = os.path.join(src, ".gitignore")
+    if os.path.exists(ignore_patterns_file):
+        rsync_cmd += "--exclude-from={} ".format(ignore_patterns_file)
+    rsync_cmd += src + "/. " + dest
+
+    cmd = ["flock", "--timeout", "600", dest, "--command", rsync_cmd]
```

Comment:
This little maneuver is costing us 1 minute 30 seconds for a new DVSim invocation in a new branch and 15 seconds for repeat invocation in the same branch.

Reply:
Well, the fault probably lies with the design of Google's IT infrastructure, so I'm not sure whether the OpenTitan GitHub project is the place to moan :-)

More constructively(!), I wonder whether we could be more selective about what gets copied? We could even have a per-IP-block (and per-tool-directory) whitelist. That could probably speed stuff up by quite a lot. It probably won't be to onerous to keep them up to date. If a whitelist casts its net too wide, Google engineers will, rightfully, complain. If it's missing some files, the nightly tests will fail.

Reply:
I agree it seems a bit too much to copy the entire repo regardless of what is needed.
but I also think as long as this has a "Google Only"  switch I don't really care.

Reply:
You're right, I did suggest it. It's the only work-around I can think of for a situation where you have to work on two disjoint compute farms. Working on an NFS mount that's visible to both your text editor and the machine with EDA licences is another approach (used in every large silicon companies I have knowledge/experience of). Google is doing it a different way, and this is the cost of doing so.

About selective copying: I'm not sure it's that problematic. As I said, if we miss something from the list, we'll find out when the nightly test fails that day: it doesn't sound so bad to me.